openhantek/src/mainwindow.cpp

```diff
@@ -155,6 +156,14 @@ MainWindow::MainWindow(HantekDsoControl *dsoControl, DsoSettings *settings, QWid
         dsoControl->setGain(channel, mSettings->scope.gain(channel) * DIVS_VOLTAGE);
     });
     connect(voltageDock, &VoltageDock::gainChanged, dsoWidget, &DsoWidget::updateVoltageGain);
+
+    connect(voltageDock, &VoltageDock::probeGainChanged, [this, dsoControl, spec, scope](ChannelID channel, unsigned probeGain) {
+        if (channel >= spec->channels) return;
+        // TODO move to the index
+        dsoControl->setProbeGain(channel, scope->voltage[channel].probeGainSteps[scope->voltage[channel].probeStepIndex]);
```

Comment:
If you implement this as a post processing step, that uses settings directly, you don't even need to notify anything because the voltage dock applies the change directly to the settings object. And the block can be removed.